test_conformance/subgroups/test_workgroup.cpp

```diff
@@ -1021,29 +977,24 @@ struct SMASK {
                 n = ii + ns > nw ? nw - ii : ns;
                 // Produce expected masks for each work item in the subgroup
                 for (i = 0; i < n; ++i) {
-                    // l - random subgroup local id
-                    l = (int)(genrand_int32(gMTdata) & 0x7fffffff) % (d > n ? n : d);
-                    int midx = 4 * ii + 4 * i ; // take subgroup local id - index;
-                    m[midx] = (cl_int)l;        // take subgroup local id - value;
+                    cl_uint max_sub_group_size = m[2];
```

Comment:
are we sure that max_sub_group_size will not change across all subgroups ?

Reply:
Guess not. Fixed.